diff --git a/Dockerfile.dapper b/Dockerfile.dapper
index 76a93e4..7ea9b05 100644
--- a/Dockerfile.dapper
+++ b/Dockerfile.dapper
@@ -8,17 +8,18 @@ RUN apt-get update && \
     apt-get install -y gcc ca-certificates git wget curl vim less file && \
     rm -f /bin/sh && ln -s /bin/bash /bin/sh
 
-ENV GOLANG_ARCH_amd64=amd64 GOLANG_ARCH_arm=armv6l GOLANG_ARCH=GOLANG_ARCH_${ARCH} \
+ENV GOLANG_ARCH_amd64=amd64 GOLANG_ARCH_arm=armv6l GOLANG_ARCH_arm64=arm64 GOLANG_ARCH=GOLANG_ARCH_${ARCH} \
     GOPATH=/go PATH=/go/bin:/usr/local/go/bin:${PATH} SHELL=/bin/bash
 
 RUN wget -O - https://storage.googleapis.com/golang/go1.9.2.linux-${!GOLANG_ARCH}.tar.gz | tar -xzf - -C /usr/local && \
     go get github.com/rancher/trash && go get github.com/golang/lint/golint
 
-ENV DOCKER_URL_amd64=https://get.docker.com/builds/Linux/x86_64/docker-1.10.3 \
-    DOCKER_URL_arm=https://github.com/rancher/docker/releases/download/v1.10.3-ros1/docker-1.10.3_arm \
+ENV DOCKER_URL_amd64=https://download.docker.com/linux/static/stable/aarch64/docker-17.09.1-ce.tgz \
+    DOCKER_URL_arm=https://github.com/rancher/docker/releases/download/v1.11.2-ros1/docker-1.11.2_arm.tgz \
+    DOCKER_URL_arm64=https://download.docker.com/linux/static/stable/aarch64/docker-17.09.0-ce.tgz \
     DOCKER_URL=DOCKER_URL_${ARCH}
 
-RUN wget -O - ${!DOCKER_URL} > /usr/bin/docker && chmod +x /usr/bin/docker
+RUN curl -sLf ${!DOCKER_URL} | tar xvzf - -C /usr/bin --strip-components=1 docker/docker && chmod +x /usr/bin/docker
 
 ENV DAPPER_ENV REPO TAG DRONE_TAG
 ENV DAPPER_SOURCE /go/src/github.com/rancher/rke-tools/
diff --git a/package/Dockerfile b/package/Dockerfile
index c1ecb87..f2d3bc2 100644
--- a/package/Dockerfile
+++ b/package/Dockerfile
@@ -3,16 +3,16 @@ FROM nginx:1.14.0-alpine
 LABEL maintainer "Rancher Labs <support@rancher.com>"
 RUN apk -U --no-cache add bash curl wget ca-certificates \
     && mkdir -p /opt/rke-tools/bin /etc/confd \
-    && curl -sLf https://github.com/kelseyhightower/confd/releases/download/v0.16.0/confd-0.16.0-linux-amd64 > /usr/bin/confd \
+    && curl -sLf https://github.com/kelseyhightower/confd/releases/download/v0.16.0/confd-0.16.0-linux-arm64 > /usr/bin/confd \
     && chmod +x /usr/bin/confd \
-    && curl -sLf https://get.docker.com/builds/Linux/x86_64/docker-1.12.3.tgz | tar xvzf - -C /opt/rke-tools/bin --strip-components=1 docker/docker \
+    && curl -sLf https://download.docker.com/linux/static/stable/aarch64/docker-17.09.1-ce.tgz | tar xvzf - -C /opt/rke-tools/bin --strip-components=1 docker/docker \
     && chmod +x /opt/rke-tools/bin/docker \
     && apk del curl
 
 RUN mkdir -p /opt/cni/bin
-RUN wget -q -O - https://github.com/containernetworking/cni/releases/download/v0.4.0/cni-amd64-v0.4.0.tgz | tar xzf - -C /tmp
+RUN wget -q -O - https://github.com/containernetworking/cni/releases/download/v0.4.0/cni-arm64-v0.4.0.tgz | tar xzf - -C /tmp
 
-RUN wget -q -O - https://github.com/coreos/etcd/releases/download/v3.0.17/etcd-v3.0.17-linux-amd64.tar.gz | tar xzf - -C /tmp && \
+RUN wget -q -O - https://github.com/coreos/etcd/releases/download/v3.3.9/etcd-v3.3.9-linux-arm64.tar.gz | tar xzf - -C /tmp && \
     mv /tmp/etcd-*/etcdctl /usr/local/bin/etcdctl && \
     rm -rf /tmp/etcd-* && rm -f /etcd-*.tar.gz && \
     apk del wget
diff --git a/scripts/package b/scripts/package
index 64b6a3f..aa5bcb3 100755
--- a/scripts/package
+++ b/scripts/package
@@ -3,9 +3,9 @@ set -e
 
 source $(dirname $0)/version
 
-ARCH=${ARCH:-"amd64"}
+ARCH=${ARCH:-"arm64"}
 SUFFIX=""
-[ "${ARCH}" != "amd64" ] && SUFFIX="_${ARCH}"
+[ "${ARCH}" != "arm64" ] && SUFFIX="_${ARCH}"
 
 cd $(dirname $0)/../package
 
